iceoryx2-cal/src/static_storage/file.rs

```diff
@@ -213,10 +214,20 @@ impl crate::named_concept::NamedConceptMgmt for Storage {
     fn list_cfg(config: &Configuration) -> Result<Vec<FileName>, NamedConceptListError> {
         let msg = "Unable to list all storages";
         let origin = "static_storage::File::list_cfg()";
-        let directory = fail!(from origin, when Directory::new(&config.path),
-            map DirectoryOpenError::InsufficientPermissions => NamedConceptListError::InsufficientPermissions,
-            unmatched NamedConceptListError::InternalError,
-            "{} due to a failure while reading the storage directory (\"{}\").", msg, config.path);
+        let directory = match Directory::new(&config.path) {
+            Ok(directory) => directory,
+            Err(DirectoryOpenError::InsufficientPermissions) => {
+                debug!(from origin, "{} due to insufficient permissions to read the storage directory.", msg);
+                return Err(NamedConceptListError::InsufficientPermissions);
+            }
+            Err(DirectoryOpenError::DoesNotExist) => {
+                return Ok(vec![]);
+            }
+            Err(v) => {
+                debug!(from origin, "{} due to failure ({:?}) while reading the storage directory (\"{}\").", msg, v, config.path);
+                return Err(NamedConceptListError::InternalError);
+            }
+        };
 
```

Comment:
Btw, the `for entry in &entries` loop could be rewritten with iterators like this
```rs
let result = entries
    .iter()
    .filter(|entry| {
        let metadata = entry.metadata();
        metadata.file_type() == FileType::File && metadata.permission() == FINAL_PERMISSIONS
    })
    .filter_map(|entry| config.extract_name_from_file(entry.name()))
    .collect();
```